pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/MailboxSendOperator.java

```diff
@@ -147,12 +150,12 @@ protected TransferableBlock getNextBlock() {
     }
   }
 
-  private void sendTransferableBlock(TransferableBlock block)
+  private boolean sendTransferableBlock(TransferableBlock block)
       throws Exception {
-    _exchange.send(block);
     if (LOGGER.isDebugEnabled()) {
       LOGGER.debug("==[SEND]== Block " + block + " sent from: " + _context.getId());
     }
+    return _exchange.send(block);
```

Comment:
(minor) Don't change the order. Cache the result because the logger should log after the block is sent

Reply:
hmm. i thought this was a bug b/c the word here is SEND not SENT :-/

Reply:
`SEND` is the name of the operator lol